Tasks/ServiceFabricComposeDeploy/ServiceFabricComposeDeploy.ps1

```diff
@@ -22,25 +22,68 @@ try {
     $removeTimeoutSec = Get-VstsInput -Name removeTimeoutSec
     $getStatusTimeoutSec = Get-VstsInput -Name getStatusTimeoutSec
 
-    $deployParameters = @{
-        'ApplicationName' = $applicationName
-        'Compose' = $composeFilePath
-    }
-    $removeParameters = @{
-        'Force' = $true
-    }
-    $getStatusParameters = @{
-        'ApplicationName' = $applicationName
-    }
-
-    $usePreviewApi = $false
-    $regKey = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Service Fabric\' -ErrorAction SilentlyContinue
+    $apiVersion = '2.8'
+    $regKey = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Service Fabric SDK\' -ErrorAction SilentlyContinue
     if ($regKey)
     {
-        if ($regKey.FabricVersion.StartsWith('255.255'))
+        if ($regKey.FabricSDKVersion -match "^\d+\.\d+")
```

Comment:
Is FabricSDKVersion key available even in older versions of service fabric sdk?

Reply:
It is available in every version of the SDK currently in support. I believe it is present in the past ones as well, but haven't personally verified

Reply:
Sure, my question was because earlier the key name we were using to find the version was different

Reply:
The original check had a bad assumption that the Runtime and SDK version would align. While this is generally true, it is not guaranteed, the SDK just has to be newer than the runtime. The cmdlets come form the SDK, so that is the version number that maters.